newIDE/app/src/ObjectEditor/Editors/SpriteEditor/PointsEditor/PointRow.js

```diff
@@ -12,90 +12,146 @@ import Text from '../../../../UI/Text';
+      setPointYInputValue(pointY.toString());
+    },
+    [pointY]
+  );
+
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <TableRow
+          style={{
+            backgroundColor: props.selected
+              ? muiTheme.listItem.selectedBackgroundColor
+              : muiTheme.list.itemsBackgroundColor,
+          }}
+          onClick={() => props.onClick(props.pointName)}
+          onMouseEnter={() => props.onMouseEnter(props.pointName)}
+          onMouseLeave={onLeave}
+        >
+          <TableRowColumn style={styles.nameColumn}>
             <TextField
               margin="none"
-              value={props.pointX}
-              type="number"
-              id="point-x"
-              onChange={(e, value) =>
-                props.onChangePointX(parseFloat(value || 0))
+              inputStyle={
+                props.selected
+                  ? { color: muiTheme.listItem.selectedTextColor }
+                  : undefined
               }
-            />
-          ) : (
-            <Text noMargin>
-              <Trans>(auto)</Trans>
-            </Text>
-          )}
-        </TableRowColumn>
-        <TableRowColumn style={styles.coordinateColumn}>
-          {!props.isAutomatic ? (
-            <TextField
-              margin="none"
-              value={props.pointY}
-              type="number"
-              id="point-y"
-              onChange={(e, value) =>
-                props.onChangePointY(parseFloat(value || 0))
+              defaultValue={props.pointName || 'Unnamed point'}
+              id={props.pointName}
+              fullWidth
+              errorText={
+                props.nameError ? 'This name is already taken' : undefined
               }
+              disabled={!props.onBlur}
+              onBlur={props.onBlur}
             />
-          ) : (
-            <Text noMargin>
-              <Trans>(auto)</Trans>
-            </Text>
-          )}
-        </TableRowColumn>
-        <TableRowColumn style={styles.toolColumn}>
-          {!!props.onRemove && (
-            <IconButton size="small" onClick={props.onRemove}>
-              <Delete />
-            </IconButton>
-          )}
-          {!!props.onEdit && (
-            <IconButton size="small" onClick={props.onEdit}>
-              <Edit />
-            </IconButton>
-          )}
-        </TableRowColumn>
-      </TableRow>
-    )}
-  </ThemeConsumer>
-);
+          </TableRowColumn>
+          <TableRowColumn style={styles.coordinateColumn}>
+            {!props.isAutomatic ? (
+              <TextField
+                margin="none"
+                inputStyle={
+                  props.selected
+                    ? { color: muiTheme.listItem.selectedTextColor }
+                    : undefined
+                }
+                value={pointXInputValue}
+                type="number"
+                id="point-x"
+                onChange={(e, value) => {
+                  setPointXInputValue(value);
```

Comment:
Same, consider a SemiControlledTextField to allow anything to be input and have the onChange called at each keystroke, but the value entered won't be replaced until the component is blurred.